src/routes/v2/sgidAuth.ts

```diff
@@ -86,25 +90,130 @@ export class SgidAuthRouter {
+      await ResultAsync.fromPromise(
+        this.usersService.loginWithEmail(userData[0].email),
+        (error) => {
+          logger.error(
+            `Error while retrieving user info from database: ${error}`
+          )
+          return new DatabaseError()
+        }
+      )
+        .map((user) => {
+          const userInfo = {
+            isomerUserId: user.id,
+            email: user.email,
+          }
+          Object.assign(req.session, { userInfo })
+          return res.status(200).send({ userData })
+        })
+        .mapErr((error) => res.status(500).send())
+    } else {
+      // User has multiple emails, defer to user to select email
+      const csrfTokenExpiry = new Date()
+      // getTime allows this to work across timezones
+      csrfTokenExpiry.setTime(csrfTokenExpiry.getTime() + CSRF_TOKEN_EXPIRY_MS)
+      const cookieSettings = {
+        expires: csrfTokenExpiry,
+        httpOnly: true,
+        secure: isSecure,
+      }
+      const token = jwtUtils.signToken({ userData })
+      res.cookie(SGID_MULTIUSER_COOKIE_NAME, token, cookieSettings)
+      return res.status(200).send({ userData })
+    }
+  }
+
+  handleSgidMultiuserLogin: RequestHandler<
+    never,
+    void | ResponseErrorBody,
+    { email: string },
+    unknown,
+    never
+  > = async (req, res) => {
+    const { email } = req.body
+    const token = req.cookies[SGID_MULTIUSER_COOKIE_NAME]
+
+    const safeVerify = Result.fromThrowable(jwtUtils.verifyToken, (error) => {
+      logger.error(
+        `Error - invalid token for sgid multiuser login ${email}: ${error}`
+      )
+      return new SgidVerifyUserError()
+    })
+
+    await safeVerify(token)
+      .andThen((verifiedToken) => {
+        if (
+          verifiedToken &&
+          typeof verifiedToken === "object" &&
+          "userData" in verifiedToken
+        ) {
+          return ok(verifiedToken.userData)
+        }
+        logger.error(
+          `Error - token does not match expected format for sgid multiuser login ${email}`
+        )
+        return err(new SgidVerifyUserError())
+      })
+      .andThen((userData) => {
+        if (
+          Array.isArray(userData) &&
+          userData.every((item) => isPublicOfficerData(item))
+        )
+          return ok(userData as PublicOfficerData[])
+        logger.error(
+          `Error - token does not match expected format for sgid multiuser login ${email}`
+        )
+        return err(new SgidVerifyUserError())
+      })
+      .andThen((userData) => {
+        const isValidUser =
+          userData.filter((data) => data.email === email).length > 0
+        if (!isValidUser) {
+          logger.error(
+            `Error - user with emails ${userData
+              .map((data) => data.email)
+              .join(", ")} attempting to login with unverified email ${email}`
+          )
+          return err(new SgidVerifyUserError())
+        }
+        return ok("")
```

Comment:
supernit: can just return the data here or omit entirely so intention is clear

Reply:
48a15e38